packages/manager/apps/hpc-vmware-managed-vcd/src/components/vrackNetwork/VrackNetworkDatagrid.component.tsx

```diff
@@ -0,0 +1,160 @@
+import {
+  ActionMenu,
+  ColumnSort,
+  Datagrid,
+  DatagridColumnTypes,
+  useColumnFilters,
+} from '@ovh-ux/manager-react-components';
+import {
+  ApiResponse,
+  applyFilters,
+  FilterCategories,
+  FilterComparator,
+} from '@ovh-ux/manager-core-api';
+import { OdsText } from '@ovhcloud/ods-components/react';
+import { useQuery } from '@tanstack/react-query';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  mockVrackSegmentList,
+  useVcdVrackNetworkOptions,
+  VrackSegment,
+} from '@ovh-ux/manager-module-vcd-api';
+import { ODS_BUTTON_VARIANT } from '@ovhcloud/ods-components';
+import VrackNetworkDatagridSubDatagrid from './VrackNetworkDatagridSubDatagrid.component';
+import { LABELS } from '@/utils/labels.constants';
+
+export type VrackNetworkDatagridProps = {
+  id: string;
+  vdcId: string;
+};
+
+export default function VrackNetworkDatagrid({
+  id,
+  vdcId,
+}: VrackNetworkDatagridProps) {
+  const { t } = useTranslation('datacentres/vrack-network');
+  const [sorting, setSorting] = useState<ColumnSort>();
+  const { filters, addFilter, removeFilter } = useColumnFilters();
+  const [searchInput, setSearchInput] = useState('');
+
+  const vcdVrackNetworkOptions = useVcdVrackNetworkOptions(id, vdcId);
+  const { data: vrackNetworks } = useQuery({
+    ...vcdVrackNetworkOptions,
+    queryFn: () =>
+      Promise.resolve({
```

Comment:
Maybe it is because you mocked the data, but i feel it would be better to managed the loading state now to be sure your don't forget it when you add the actual fetch request